### PR TITLE
[SOAR-5441][SOAR-5440] - Cortex XDR File Actions

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Proposed Changes
This PR adds the block file and allow file actions to the cortex xdr plugin

### Block File: 
```
RMT-MBP-5357:palo_alto_cortex_xdr jmcadams$ ./run.sh -R tests/block_file.json -j
Running: cat tests/block_file.json | docker run --rm   -i rapid7/palo_alto_cortex_xdr:1.0.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Connecting...
rapid7/Palo Alto Cortex XDR:1.0.0. Step name: block_file
Taking Block action on file: 275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f
Endpoint: /public_api/v1/hash_exceptions/blocklist/
Starting new HTTPS connection (1): api-rapid7.xdr.us.paloaltonetworks.com:443
https://api-rapid7.xdr.us.paloaltonetworks.com:443 "POST /public_api/v1/hash_exceptions/blocklist/ HTTP/1.1" 200 35
Block action was successful
Connect: Connecting...
rapid7/Palo Alto Cortex XDR:1.0.0. Step name: block_file
Taking Block action on file: 275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f
Endpoint: /public_api/v1/hash_exceptions/blocklist/
Block action was successful

{
  "success": true
}
```

### Allow File:
```
RMT-MBP-5357:palo_alto_cortex_xdr jmcadams$ ./run.sh -R tests/allow_file.json -j
Running: cat tests/allow_file.json | docker run --rm   -i rapid7/palo_alto_cortex_xdr:1.0.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Connecting...
rapid7/Palo Alto Cortex XDR:1.0.0. Step name: allow_file
Taking Allow action on file: 275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f
Endpoint: /public_api/v1/hash_exceptions/allowlist/
Starting new HTTPS connection (1): api-rapid7.xdr.us.paloaltonetworks.com:443
https://api-rapid7.xdr.us.paloaltonetworks.com:443 "POST /public_api/v1/hash_exceptions/allowlist/ HTTP/1.1" 200 35
Allow action was successful
Connect: Connecting...
rapid7/Palo Alto Cortex XDR:1.0.0. Step name: allow_file
Taking Allow action on file: 275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f
Endpoint: /public_api/v1/hash_exceptions/allowlist/
Allow action was successful

{
  "success": true
}
```

### CODE REVIEW

File: palo_alto_cortex_xdr/help.md [hunk 1]
```diff
@@ -53,6 +53,78 @@ Example input:
...
+
+This action is used to add a file to the block list.
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|comment|string|File blocked by InsightConnect|True|String that represents additional information regarding the action|None|File blocked by InsightConnect|
+|file_hash|string|None|True|A SHA256 file hash|None|275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f|
+|incident_id|integer|None|False|If this is related to an incident, the ID should be entered here|None|5|
```
Comment (mrinehart-r7) [inline palo_alto_cortex_xdr/help.md:66]:
```suggestion
|incident_id|string|None|False|If this is related to an incident, the ID should be entered here|None|5|
```

File: palo_alto_cortex_xdr/help.md [hunk 2]
```diff
@@ -53,6 +53,78 @@ Example input:
...
+
+This action is used to add a file to the allow list.
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|comment|string|File allowed by InsightConnect|True|String that represents additional information regarding the action|None|File allowed by InsightConnect|
+|file_hash|string|None|True|A SHA256 file hash|None|275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f|
+|incident_id|integer|None|False|If this is related to an incident, the ID should be entered here|None|5|
```
Comment (mrinehart-r7) [inline palo_alto_cortex_xdr/help.md:102]:
```suggestion
|incident_id|string|None|False|If this is related to an incident, the ID should be entered here|None|5|
```

File: palo_alto_cortex_xdr/help.md [hunk 3]
```diff
@@ -53,6 +53,78 @@ Example input:
...
+
+Example input:
+
+```
+{
+  "comment": "File allowed by InsightConnect",
+  "file_hash": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
+  "incident_id": 5
```
Comment (mrinehart-r7) [inline palo_alto_cortex_xdr/help.md:110]:
```suggestion
  "incident_id": "5"
```

File: palo_alto_cortex_xdr/help.md [hunk 4]
```diff
@@ -53,6 +53,78 @@ Example input:
...
+|file_hash|string|None|True|A SHA256 file hash|None|275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f|
+|incident_id|integer|None|False|If this is related to an incident, the ID should be entered here|None|5|
+
+Example input:
+
+```
+{
+  "comment": "File blocked by InsightConnect",
+  "file_hash": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
+  "incident_id": 5
```
Comment (mrinehart-r7) [inline palo_alto_cortex_xdr/help.md:74]:
```suggestion
  "incident_id": "5"
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py [hunk 5]
```diff
@@ -153,6 +153,33 @@ def get_incidents(self, from_time, to_time, time_field="creation_time"):
...
 
+    def allow_or_block_file(self, file_hash, comment, incident_id=None, block_file=True):
+        if block_file:
+            endpoint = "/public_api/v1/hash_exceptions/blocklist/"
+        else:
+            endpoint = 	"/public_api/v1/hash_exceptions/allowlist/"
+
+        post_body = {
+            "request_data": {
+                "hash_list": [file_hash],
```
[the remark this answers is not in the document]
Comment (jmcadams-r7) [reply palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py:164]: I went for ease of use over matching the API. I think arrays for this sort of thing are confusing.

**Pull request merged**